src/test/librados/service.cc

```diff
@@ -45,6 +48,118 @@ TEST(LibRadosService, RegisterLate) {
   rados_shutdown(cluster);
 }
 
+static void status_format_func(const int i, std::mutex &lock,
+                               std::condition_variable &cond,
+                               int &threads_started, bool &stopped)
+{
+  rados_t cluster;
+  char *metadata_buf = NULL;
+
+  ASSERT_EQ(0, rados_create(&cluster, "admin"));
+  ASSERT_EQ(0, rados_conf_read_file(cluster, NULL));
+  ASSERT_EQ(0, rados_conf_parse_env(cluster, NULL));
+
+  ASSERT_EQ(0, rados_connect(cluster));
+  string tid = stringify(std::this_thread::get_id());
+  string prefix = string("gw") + stringify(i % 8);
+  ASSERT_NE(-1, asprintf(&metadata_buf, "%s%c%s%c%s%c%s%c",
+                         "daemon_type", '\0', "portal", '\0',
+                         "daemon_prefix", '\0', prefix.c_str(), '\0'));
+  string name = string("rbd/image") + tid;
+  ASSERT_EQ(0, rados_service_register(cluster, "iscsi", name.c_str(),
+		                      metadata_buf));
+
+  std::unique_lock<std::mutex> l(lock);
+  threads_started++;
+  cond.notify_all();
+  cond.wait(l, [&stopped] {
+    return stopped;
+  });
+
+  rados_shutdown(cluster);
+}
+
+TEST(LibRadosService, StatusFormat) {
+  const int nthreads = 128;
```

Comment:
Nit: probably no need to create 128+ connections to the cluster to validate the functionality.

Reply:
Okay, will use 16 instead.